### DISCUSSION TITLE
How to have a multiline button?

### DISCUSSION THREAD

**Discussion by EduVencovsky** [callstack/react-native-paper#2584] [Q&A]
I'm trying to use the `Button` component with multiple lines of text, is it even possible?

I was looking at the source code and found that the `Text` component of the `Button` have a fixed prop `numberOfLines={1}`. Maybe it should be possible to customize the `Text` component passing something like `textProps`?

I know that multiline button goes agains material design guide lines, but... is it possible?

**Comment by raajnadar** [comment]
Did you try

```
<Button>
   <Text>Multiline test</Text>
   <Text>Did that work?</Text>
</Button>
```

**Reply by latobibor** [reply]
This would not work when you don't know the exact text or when you deal with localized text which can be either very short for one language and very long for another.

**Comment by EduVencovsky** [comment]
@raajnadar doesn't work because nested `Text` components are treated as only one "text" for react native

**Comment by peterchibunna** [comment]
If you want a multiline text Button, use the TouchableRipple component wrapped with the Surface component:
```
<Surface ...>
  <TouchableRipple ...>
     <Text numberOfLines={2}>
      ...
      </Text>
   </TouchableRipple>
</Surface>
```

and then apply the button theme of your app or the DefaultTheme to the `Surface` component.

**Reply by EduVencovsky** [reply]
> apply the button theme of your app or the DefaultTheme to the Surface component

How would that be?

**Reply by peterchibunna** [reply]
Basically style the `Surface` as you want, because Surface has `theme.colors.surface` background color by default and button has `theme.colors.primary` as the default background color

**Comment by bcye** [comment]
If you want to keep using a Button component - use an IconButton component and set the Text as a custom button, though you will have to use a fixed width, else it will have the default icon width. You could also use the icon prop on a normal button, but it has some stylings applied to it that make the IconButton a better candidate:
```jsx
<IconButton
  style={{ width: "28%", maxHeight: "100%" }}
  icon={() => (
  <Text
    style={{
      width: "100%",
      textAlign: "center",
    }}
    numberOfLines={2}
  >
    Your multiline text
  </Text>
  )}
/>
```

**Comment by adbs1** [comment]
Why not just make numberOfLines configurable?

The first is ugly, but unlikely to happen, still very usable, and more accessible than an ellipsis if it does happen. I need the "press 'n' hold", and it is easily supported IF the numberOfLines is configurable.

<img width="430" alt="image" src="https://github.com/user-attachments/assets/39f304a3-da5b-4ce9-9dc6-e4aca6dfa349">
<img width="156" alt="image" src="https://github.com/user-attachments/assets/61ff1817-1a2e-4d91-a23e-82f1bb7bfabe">
